packages/flutter/lib/src/material/pickers/calendar_date_picker.dart

```diff
@@ -615,11 +774,35 @@ class _MonthPickerState extends State<_MonthPicker> {
   }
 }
 
+/// InheritedWidget indicating what the current focused date is for its children.
+///
+/// This is used by the [_MonthPicker] to let its children [_DayPicker]s what
+/// the currently focused date (if any) should be.
+class _FocusedDate extends InheritedWidget {
+  const _FocusedDate({
+    Key key,
+    Widget child,
+    this.date
+  }) : super(key: key, child: child);
+
+  final DateTime date;
+
+  @override
+  bool updateShouldNotify(_FocusedDate oldWidget) {
+    return date != oldWidget.date && !utils.isSameDay(date, oldWidget.date);
```

Comment:
When would both `date != oldWidget.date` and `utils.isSameDay(date, oldWIdge.date)` both be true?

Reply:
It is `!utils.isSameDay(date, oldWidget.date)`. It should return true if the date is different than oldWidget.date. This is a convoluted way of saying that though, so I moved some null checking into the `isSameDay` code to simplify this to just:

```dart
return !utils.isSameDay(date, oldWidget.date);
```